src/convert.cpp

```diff
@@ -113,26 +121,37 @@ ddwaf_object* to_ddwaf_object(
   }
   if (val.IsNumber()) {
     mlog("creating Number");
-    return ddwaf_object_signed(object, val.ToNumber().Int64Value());
+    if (coerceToString) {
+      return ddwaf_object_string(object, val.ToString().Utf8Value().c_str());
+    } else {
+      return ddwaf_object_signed(object, val.ToNumber().Int64Value());
+    }
   }
   if (val.IsBoolean()) {
     mlog("creating Boolean");
-    if (coerceBoolToInt) {
-      mlog("turning boolean to int");
-      int64_t nb = val.ToBoolean().Value() ? 1 : 0;
-      return ddwaf_object_signed(object, nb);
+    if (coerceToString) {
+      return ddwaf_object_string(object, val.ToString().Utf8Value().c_str());
```

Comment:
Maybe you could optimize a tiny bit by keeping the ternary like `val.ToBoolean().Value() ? "true" : "false";` wdyt ?

Reply:
because we know that's the only possible values for a bool, no need for a toString() right ?